File: 605. Can Place Flowers.py
class Solution:
    def canPlaceFlowers(self, flowerbed: List[int], n: int) -> bool:
        canAdd = 0

        for i in range(len(flowerbed)):
            left = flowerbed[i - 1] if i > 0 else 0
            right = flowerbed[i + 1] if i < len(flowerbed) - 1 else 0

            if left == 0 and flowerbed[i] == 0 and right == 0:
                flowerbed[i] = 1
                canAdd += 1

        return n <= canAdd
        

File: 605. Can Place Flowers.js
/**
 * @param {number[]} flowerbed
 * @param {number} n
 * @return {boolean}
 */
var canPlaceFlowers = function (flowerbed, n) {
  let canAdd = 0;

  for (let i = 0; i < flowerbed.length; i++) {
    const left = flowerbed[i - 1] ?? 0;
    const right = flowerbed[i + 1] ?? 0;

    if (flowerbed[i] === 0 && left === 0 && right === 0) {
      flowerbed[i] = 1;
      canAdd++;
    }
  }

  return n <= canAdd;
};
